File: src/main/java/me/yoursole/ctf/Events/DropItemEvent.java
package me.yoursole.ctf.Events;

import me.yoursole.ctf.DataFiles.Items.Flag;
import org.bukkit.ChatColor;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.player.PlayerDropItemEvent;

public class DropItemEvent implements Listener {
    @EventHandler
    public void onItemDrop(PlayerDropItemEvent e){
        if(e.getItemDrop().getItemStack().isSimilar(Flag.flag)){
            e.setCancelled(true);
            e.getPlayer().sendMessage(ChatColor.RED+"You can not get rid of this");
        }

    }
}

File: src/main/kotlin/me/yoursole/ctf/events/DropItemEvent.kt
package me.yoursole.ctf.events

import me.yoursole.ctf.datafiles.items.Flag
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.player.PlayerDropItemEvent

object DropItemEvent : Listener {
    @EventHandler
    fun onItemDrop(e: PlayerDropItemEvent) {
        println("1")
        if (e.itemDrop.itemStack.itemMeta == Flag.flag.itemMeta) {
            println("2")
            e.isCancelled = true
            e.player.sendMessage("§cYou can not get rid of this")
        }
    }
}